src/pages/Admin.js

```diff
@@ -34,37 +36,321 @@ const RiseUp = styled.div`
+  height: 15px;
+  background-color: red;
+  mask: url(${closeImg}) center no-repeat;
+  cursor: pointer;
+`;
+
+const Code = styled.pre`
+  font-size: 14px;
+  padding: 30px;
 `;
 
+const ABSTAIN = 'Abstain';
+const NOCHANGE = 'No Change';
+
 class Admin extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      pollTitle: '',
+      pollSummary: '',
+      pollStart: null,
+      pollEnd: null,
+      pollLink: '',
+      pollOption: '',
+      pollOptions: [ABSTAIN, NOCHANGE],
+      pollContent: '',
+      pollMarkdown: '',
+      pollGenerated: false
+    };
+  }
+
+  handlePollState = (e, k) => {
+    e.stopPropagation();
+    this.setState({
+      ...this.state,
+      [k]: e.target.value
+    });
+  };
+
+  addPollVoteOption = () => {
+    if (this.state.pollOption.length) {
+      const opt = this.state.pollOption;
+      const opts = this.state.pollOptions;
+      this.setState({
+        pollOption: '',
+        pollOptions: [...opts, opt]
+      });
+    }
+  };
+
+  removePollOption = idx => {
+    const opts = this.state.pollOptions.filter((item, index) => index !== idx);
+    this.setState({
+      pollOptions: opts
+    });
+  };
+
+  resetPollState = () => {
+    this.setState({
+      pollTitle: '',
+      pollSummary: '',
+      pollLink: '',
+      pollOption: '',
+      pollOptions: [ABSTAIN, NOCHANGE],
+      pollContent: ''
+    });
+  };
+
+  parseFormToMarkdownString = () => {
+    const optsString = this.state.pollOptions.reduce(
+      (acc, opt, idx) => `${acc}   ${idx}: ${opt}\n`,
+      ''
+    );
+    const yml = `---\ntitle: ${this.state.pollTitle}\nsummary: ${
+      this.state.pollSummary
+    }\ndiscussion_link: ${this.state.pollLink}\noptions:\n${optsString}---\n`;
+    const md = `# Poll: ${this.state.pollTitle}\n\n${this.state.pollContent}`;
+    this.setState({
+      pollMarkdown: `${yml}${md}`,
+      pollGenerated: true
+    });
+  };
+
   render = () => {
+    const isValidSubmission =
+      !!this.state.pollTitle &&
+      !!this.state.pollSummary &&
+      this.state.pollOptions.length > 2 &&
+      !!this.state.pollContent;
+    // this.pollStart &&
+    // this.pollEnd &&
+
     return (
-      <RiseUp>
-        <StyledTop>
-          <StyledTitle>Create Polling Proposal</StyledTitle>
-        </StyledTop>
-        <ContentWrapper>
-          <ProposalPanel>Full proposal goes here</ProposalPanel>
-          <RightPanels>Other input goes here</RightPanels>
-        </ContentWrapper>
-      </RiseUp>
+      <Fragment>
+        {this.state.pollGenerated ? (
+          <RiseUp>
+            <StyledTop>
+              <StyledTitle>Create a new Polling proposal</StyledTitle>
+            </StyledTop>
+            <ContentWrapper>
+              <Code>{this.state.pollMarkdown}</Code>
+              <SectionWrapper>
+                <Button
+                  slim
+                  onClick={() =>
+                    navigator.clipboard.writeText(this.state.pollMarkdown)
```

Comment:
we also have [this](https://github.com/makerdao/governance-dashboard/blob/a90606d13883a430b6842d1eb0c000378892df42/src/utils/misc.js#L206) util fnc to copy text to the clipboard, though if this one works, i say keep it